sbin/spark-daemon.sh

```diff
@@ -162,6 +162,15 @@ run_command() {
   esac
 
   echo "$newpid" > "$pid"
+
+  for i in {1..10}
+  do
+    if [[ $(ps -p "$newpid" -o comm=) =~ "java" ]]; then
+       break
+    fi
+    sleep 0.5
```

Comment:
Oh..the sleep 2 statement can't move.

the loop may take at most 5 seconds to check whether the daemon turn into java daemon. When it become java daemon, it jump out the loop.

**then must sleep 2 seconds,**
then check whether the java daemon die. Because java daemon may exist for some time and then throw an Exception and die. So here need to wait for some time.